### PR TITLE
MTDSA-7754 update dependencies

### PR DESCRIPTION
**Pull request by yangwong9664**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -22,19 +22,19 @@ object AppDependencies {
 
   val compile: Seq[ModuleID] = Seq(
     ws,
-    "uk.gov.hmrc" %% "bootstrap-backend-play-27" % "2.20.0",
-    "uk.gov.hmrc"   %% "domain"            % "5.9.0-play-27",
+    "uk.gov.hmrc" %% "bootstrap-backend-play-27" % "3.2.0",
+    "uk.gov.hmrc"   %% "domain"            % "5.10.0-play-27",
     //"uk.gov.hmrc"   %% "play-hmrc-api"     % "4.1.0-play-26",
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:27]: Can we please just get rid of this as it's not needed. Thanks

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -22,19 +22,19 @@ object AppDependencies {
...
-    "org.typelevel" %% "cats-core"         % "2.2.0",
+    "org.typelevel" %% "cats-core"         % "2.3.0",
     compilerPlugin("com.github.ghik" % "silencer-plugin" % "1.7.1" cross CrossVersion.full),
     "com.github.ghik" % "silencer-lib" % "1.7.1" % Provided cross CrossVersion.full,
     "com.chuusai"   %% "shapeless"         % "2.4.0-M1"
   )
 
   def test(scope: String = "test, it"): Seq[sbt.ModuleID] = Seq(
-    "org.scalatest"          %% "scalatest"          % "3.2.2"             % scope,
+    "org.scalatest"          %% "scalatest"          % "3.2.3"             % scope,
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:35]: scalatest should be bumped to 3.3.0-SNAP2. Also jettyFromWiremockVersion should be bumped to 9.4.35.v20201120. The one we are using currently is 2018 version.

**Review by Taiwo-AB — COMMENTED** [review]
Also, these plugins.sbt needs updating. These are the following versions. 
uk.gov.hmrc -> sbt-auto-build -> 2.11.0
uk.gov.hmrc -> sbt-git-versioning -> 2.2.0
uk.gov.hmrc -> sbt-artifactory -> 1.10.0
uk.gov.hmrc -> sbt-distributables -> 2.1.0
com.typesafe.play -> sbt-plugin -> 2.7.5
org.scalastyle -> scalastyle-sbt-plugin -> 1.0.0
org.scoverage -> sbt-scoverage -> 1.6.1
com.timushev.sbt -> sbt-updates -> 0.5.1

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -22,19 +22,19 @@ object AppDependencies {
...
     "com.vladsch.flexmark"   % "flexmark-all"        % "0.36.8"           % scope,
-    "org.scalacheck"         %% "scalacheck"         % "1.14.3"            % scope,
+    "org.scalacheck"         %% "scalacheck"         % "1.15.1"            % scope,
     "org.scalamock"          %% "scalamock"          % "5.0.0"             % scope,
```
Comment (Taiwo-AB) [inline project/AppDependencies.scala:38]: This should be 5.1.0

**Review by Taiwo-AB — CHANGES_REQUESTED** [review]
Also, these plugins.sbt needs updating. These are the following versions.
uk.gov.hmrc -> sbt-auto-build -> 2.11.0
uk.gov.hmrc -> sbt-git-versioning -> 2.2.0
uk.gov.hmrc -> sbt-artifactory -> 1.10.0
uk.gov.hmrc -> sbt-distributables -> 2.1.0
com.typesafe.play -> sbt-plugin -> 2.7.5
org.scalastyle -> scalastyle-sbt-plugin -> 1.0.0
org.scoverage -> sbt-scoverage -> 1.6.1
com.timushev.sbt -> sbt-updates -> 0.5.1

File: project/AppDependencies.scala (hunk 2 above)
Comment (yangwong9664) [reply project/AppDependencies.scala:35]: we don't typically upgrade to non-main releases, like snapshots etc. I'll keep scalatest at 3.2.3

**Pull request merged**